ext/cuda/bilinearinterpolation.jl

```diff
@@ -0,0 +1,169 @@
+import ..ClimaInterpolations.BilinearInterpolation:
+    Bilinear,
+    set_source_range!,
+    interpolatebilinear!,
+    get_stencil_bilinear1d,
+    get_dims
+
+function interpolatebilinear!(
```

Comment:
Could you add a docstring please? Commenting on e.g. how is this function different from the non-CUDA function in src?